### DISCUSSION TITLE
Nested items

### DISCUSSION THREAD

**Discussion by sultanmyrza** [flutterdata/flutter_data#124] [Q&A] [answered]
Hello, 
I'm trying to use your library for my case. I read the documents but still not sure will this package work for me?

# My case
Here is my case. I'm basically trying to build simple version of iCloud.
I need to do CRUD operations with `items` object (by item I refer to `file` or `directory` in my case).

Here are examples responses
**GET request root directory ** `https://api.myserver.com/api/v1/files/users/userId?limit=30&offset=0&direction=asc`
 ```json
{
    "data": [
        {
            "id": "1dd379dd-e9f1-49d5-8d57-01c2f231fd5d",
            "type": "FILE",
            "filename": "3.PNG",
            "preview": "{{ preview_url }}",
            "uploadUrl": "{{ upload_url }}"
        },
        {
            "id": "613d1e22-f82a-49b7-86c1-05e1ff4fc44a",
            "type": "FILE",
            "filename": "31.PNG",
            "preview": "{{ preview_url }}",
            "uploadUrl": "{{ upload_url }}"
        },
        {
            "id": "7e4edba5-9644-4e7e-b991-3b5d4ac53b8f",  <--- subDirectory
            "type": "DIRECTORY",
            "filename": "my directory"
        }
    ]
}
```
**GET request subdirectory** `https://api.myserver.com/api/v1/dir/{{subDirerectoryId}}/users/{{userId}}?limit=30&offset=0&direction=asc`
> in this case subDirectoryId = 7e4edba5-9644-4e7e-b991-3b5d4ac53b8f

```json
{
    "data": [
        {
            "id": "0966c025-2894-4d33-849f-e5aae9abedf1",
            "type": "FILE",
            "filename": "MY MOVIE.MP4",
            "preview": "{{ preview_url }}",
            "uploadUrl": "{{ upload_url }}"
        },
        {
            "id": "31ebc4c8-105b-422e-a106-0b14fcb4db67",
            "type": "FILE",
            "filename": "MY CODE.ZIP",
            "preview": "{{ preview_url }}",
            "uploadUrl": "{{ upload_url }}"
        },
        {
            "id": "117f89da-1929-48ea-985b-d8384724b9b7",
            "type": "DIRECTORY",
            "filename": "another directory"
        },
        {
            "id": "c8c191b9-4c78-407d-b826-f07550357415",
            "type": "DIRECTORY",
            "filename": "my vocation"
        }
    ]
}
```
# My Questions

I read about [relationships](https://flutterdata.dev/docs/relationships/)
> Remove a relationship
> Given a Post with many Comments we want to remove:
> 
> final postWithNoComments = post.copyWith(comments: HasMany.remove()).was(post);
> Works with both HasMany and BelongsTo.
> 
> Removing a relationship does not delete its linked resources (the actual comments in this case).
## Question 1.
If removing a relationship does not delete its linked resources what can I do? For now I see one solution.

Which is to loop through post.comments and delete each comment. But I'm not sure what will happen inside `flutter_data`
will it updated.

**Also wondering what will do flutter_data when I will move multiplie files to another sub directory?**

# My final thoughs
> P.S: I think this package is good enough 👍. 
> If this packages works for my case I would be happy to contribute or just buy a coffee 😅.

**Comment by frank06** [comment, marked as answer]
It's definitely possible. Deleting can be done as you describe. Moving is simply using relationships to link to another resource. Tree-like structures are also possible. There are tests showing all of these capabilities.

**Reply by sultanmyrza** [reply]
Thank you for your reply 🤗
